### PR TITLE
fix: replace 0 with MaxInt when upgrading query-concurrency

### LINKED ISSUE #19927: config.toml file contains invalid value for query-concurrency
when using the default influxdb 1.x config options, the upgrade process generates a new config for 2.x.

there is an option for query-concurrency which must be > 0

in 1.x, `max-concurrent-queries = 0` means infinite, but in 2.x, you need to use MAXINT to get infinite. 

```
➜  influxdb-2.0.0-rc.4_darwin_amd64 INFLUXD_CONFIG_PATH=../config.toml ./influxd
2020-11-06T21:31:54.993587Z	info	Welcome to InfluxDB	{"log_id": "0QJsPxxG000", "version": "2.0.0-rc.4", "commit": "0ae3df0f27", "build_date": "2020-11-06T02:08:27Z"}
2020-11-06T21:31:55.011470Z	info	Resources opened	{"log_id": "0QJsPxxG000", "service": "bolt", "path": "/Users/rsavage/workspace/influxdb_testing/.influxdbv2/influxd.bolt"}
2020-11-06T21:31:55.146232Z	info	Checking InfluxDB metadata for prior version.	{"log_id": "0QJsPxxG000", "bolt_path": "/Users/rsavage/workspace/influxdb_testing/.influxdbv2/influxd.bolt"}
2020-11-06T21:31:55.146513Z	info	Using data dir	{"log_id": "0QJsPxxG000", "service": "storage-engine", "path": "/Users/rsavage/workspace/influxdb_testing/.influxdbv2/engine/data"}
2020-11-06T21:31:55.146603Z	info	Compaction settings	{"log_id": "0QJsPxxG000", "service": "storage-engine", "max_concurrent_compactions": 4, "throughput_bytes_per_second": 50331648, "throughput_bytes_per_second_burst": 50331648}
2020-11-06T21:31:55.146622Z	info	Open store (start)	{"log_id": "0QJsPxxG000", "service": "storage-engine", "op_name": "tsdb_open", "op_event": "start"}
2020-11-06T21:31:55.269110Z	info	Opened shard	{"log_id": "0QJsPxxG000", "service": "storage-engine", "op_name": "tsdb_open", "index_version": "inmem", "path": "/Users/rsavage/workspace/influxdb_testing/.influxdbv2/engine/data/080bbc253b44dae3/autogen/2", "duration": "45.889ms"}
2020-11-06T21:31:55.269168Z	info	Open store (end)	{"log_id": "0QJsPxxG000", "service": "storage-engine", "op_name": "tsdb_open", "op_event": "end", "op_elapsed": "122.767ms"}
2020-11-06T21:31:55.269178Z	info	Starting retention policy enforcement service	{"log_id": "0QJsPxxG000", "service": "retention", "check_interval": "30m"}
2020-11-06T21:31:55.269195Z	info	Starting precreation service	{"log_id": "0QJsPxxG000", "service": "shard-precreation", "check_interval": "10m", "advance_period": "30m"}
2020-11-06T21:31:55.269260Z	error	Failed to create query controller	{"log_id": "0QJsPxxG000", "error": "invalid controller config: ConcurrencyQuota must be positive"}
Error: invalid controller config: ConcurrencyQuota must be positive
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #19927 

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influxd/upgrade/config.go [hunk 1]
```diff
@@ -45,12 +61,22 @@ var configMapRules = map[string]string{
 	"http.https-private-key":                               "tls-key",
 }
 
+// configValueTransforms is a map from 2.x config keys to transformation functions
+// that should run on the 1.x values before they're written into the 2.x config.
+//
+// NOTE: This might be overkill for only 1 key, I wrote this before we had fully
+// investigated which config keys needed to be transformed in this way.
+var configValueTransforms = map[string]func(interface{}) interface{}{
+	"query-concurrency": rewriteInf,
```
Comment (danxmoran) [inline cmd/influxd/upgrade/config.go:70]: The patch here is definitely overkill if we only ever end up with 1 entry in this map. I wrote the fix before we'd nailed down the full list of settings that needed to be transformed during upgrade. As far as I can tell, query-concurrency turned out to be the only one that actually needs it.

If we'd rather have a simpler patch I can replace all this with a hard-coded check for query-concurrency and 0 within the `updateV2Config` loop.

File: cmd/influxd/upgrade/config_test.go [hunk 2]
```diff
@@ -372,7 +372,7 @@ influxql-max-select-buckets = 0
 influxql-max-select-point = 0
 influxql-max-select-series = 0
 log-level = "info"
-query-concurrency = 0
+query-concurrency = 9223372036854775807
```
Comment (dgnorton) [inline cmd/influxd/upgrade/config_test.go:375]: Should this use `launcher.MaxInt` like above or `math.MaxInt`?

**Review by dgnorton — COMMENTED** [review]
Mostly looks good. One question though

File: cmd/influxd/upgrade/config_test.go (hunk 2 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/config_test.go:375]: I tried it with `max.MaxInt` to start, and the test still passed with `0` here... assumed the number was too big and something funky was happening with the parser. I'll give it another try just to sanity-check.

**Review by docmerlin — APPROVED** [review]
This looks good to me,  it feels hella hacky though.
Maybe after this release we should get it so zero or negative concurrency is treated as unlimited in the query system?

File: cmd/influxd/upgrade/config_test.go (hunk 2 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/config_test.go:375]: Looks like `launcher.MaxInt` == `math.MaxInt32`

**Pull request merged**